src/models/activityReportApprover.js

```diff
@@ -76,7 +69,47 @@ module.exports = (sequelize, DataTypes) => {
   }, {
     hooks: {
       afterCreate: async (instance, options) => {
-        // The following code should match afterUpsert.
+        // The following code should match other hooks.
+        // This can not be abstracted into a function.
+        // Begin
+        const approverStatuses = await sequelize.models.ActivityReportApprover.findAll({
+          attributes: ['status'],
+          raw: true,
+          where: { activityReportId: instance.activityReportId },
+          transaction: options.transaction,
+        }).map((a) => a.status);
+
+        const newCalculatedStatus = calculateReportStatus(instance.status, approverStatuses);
+        await sequelize.models.ActivityReport.update({
+          calculatedStatus: newCalculatedStatus,
+        }, {
+          where: { id: instance.activityReportId },
+          transaction: options.transaction,
+          hooks: false,
+        });
+        // End
+      },
+      afterDestroy: async (instance, options) => {
+        // Code unique to this hook
+        const approverStatuses = await sequelize.models.ActivityReportApprover.findAll({
```

Comment:
Yes, since we're using an "after" hook the db transaction has already been completed, `ActivityReportApprover.findAll` will not include the recently destroyed item. Here we're also careful not to use the "instance" (the destroyed record held in memory just for the after hook) to determine the report status. This is why we skip straight to using the `calculateReportStatusFromApprovals` function, rather than the `calculateReportStatus` function that most other hooks use. This does has the potential to get **very** confusing so I wrote a test that ensure this works as expected. https://github.com/adhocteam/Head-Start-TTADP/blob/sj-mult-approvers/src/services/activityReportApprovers.test.js#L130-L154 If the recently deleted approver record was still being used to create calculatedStatus, then line 149 `expect(afterDeleteReport.calculatedStatus).toEqual(REPORT_STATUSES.SUBMITTED);` would fail, the status wold still be `NEEDS_ACTION`.